packages/custom_lint/lib/custom_lint.dart

```diff
@@ -71,21 +72,32 @@ Future<void> _runPlugins(
   CustomLintRunner runner, {
   required bool reload,
 }) async {
+  final log = Logger.standard();
+  final progress = log.progress('Analyzing');
+
   try {
     final lints = await runner.getLints(reload: reload);
 
     if (lints.any((lintsForFile) => lintsForFile.errors.isNotEmpty)) {
       exitCode = 1;
     }
 
-    _renderLints(lints, workingDirectory: runner.workingDirectory);
+    _renderLints(
+      log,
+      progress,
+      lints,
+      workingDirectory: runner.workingDirectory,
+    );
   } catch (err, stack) {
     exitCode = 1;
-    stderr.writeln('$err\n$stack');
+    log.stderr(err.toString());
```

Comment:
Let's move all the rendering logic to `_renderLints`

Reply:
Sure

Reply:
Can I not just remove the try/catch here completely and move the error logging to https://github.com/invertase/dart_custom_lint/blob/main/packages/custom_lint/lib/custom_lint.dart#L62?

Reply:
I would like to take a stab at this separately as well.